app.rb

```diff
@@ -3,25 +3,47 @@
 require 'sinatra/json'
 require 'pry'
 require 'jwe'
+require 'securerandom'
+require 'uuid'
 
 PAYLOAD_CONTENT_FILE = 'tmp/payload_content.json'
 
+helpers do
+  def read_payload_file(filename)
+    encrypted_payload = File.read(filename)
+    JWE.decrypt(encrypted_payload, ENV['ENCRYPTION_KEY'])
+  rescue Errno::ENOENT
+    status 404
+    json({ error: 'Submission not found' })
+  rescue JWE::InvalidData
+    status 400
+    json(
+      { error: 'Failed to decrypt submission. Is the encryption key correct?' }
+    )
+  end
+end
+
 post '/submission' do
   payload = request.body.read
-  File.write(PAYLOAD_CONTENT_FILE, payload)
+  uuid = SecureRandom.uuid
+  ["tmp/#{uuid}", PAYLOAD_CONTENT_FILE].each do |filename|
+    File.write(filename, payload)
+  end
+  status 201
+  headers['Location'] = "/submission/#{uuid}"
 end
 
 get '/submission' do
-  encrypted_payload = File.read(PAYLOAD_CONTENT_FILE)
-  JWE.decrypt(encrypted_payload, ENV['ENCRYPTION_KEY'])
-rescue Errno::ENOENT
-  status 404
-  json({ error: 'Submission not found' })
-rescue JWE::InvalidData
-  status 400
-  json(
-    { error: 'Failed to decrypt submission. Is the encryption key correct?' }
-  )
+  read_payload_file(PAYLOAD_CONTENT_FILE)
+end
+
+get '/submission/:id' do
+  if UUID.validate(params[:id])
+    read_payload_file("tmp/#{params[:id]}")
+  else
+    status 400
+    json({ error: 'Submission id should be a valid UUID' })
+  end
```

Comment:
since we're adding to them, shall we use the rails `:symbols`, `:created` etc?

Reply:
Does Sinatra support the symbols? I didn't know :)

Reply:
oh sorry! i completely forgot it's not a rails app 😆

Reply:
It doesn't. I mark as resolved if that's okay :)